lib/maths/CTimeSeriesModel.cc

```diff
@@ -973,137 +973,163 @@ bool CUnivariateTimeSeriesModel::probability(const CModelProbabilityParams& para
     if (value.empty()) {
         return true;
     }
+    return value[0].size() == 1
+               ? this->uncorrelatedProbability(params, time, value, probability, tail)
+               : this->correlatedProbability(params, time, value, probability, tail,
+                                             conditional, mostAnomalousCorrelate);
+}
 
-    if (value[0].size() == 1) {
-        core_t::TTime time{time_[0][0]};
-        TDouble1Vec sample{m_TrendModel->detrend(
-            time, value[0][0], params.seasonalConfidenceInterval())};
-        maths_t::TDoubleWeightsAry1Vec weights{unpack(params.weights()[0])};
+bool CUnivariateTimeSeriesModel::uncorrelatedProbability(const CModelProbabilityParams& params,
+                                                         const TTime2Vec1Vec& time_,
+                                                         const TDouble2Vec1Vec& value,
+                                                         double& probability,
+                                                         TTail2Vec& tail) const {
```

Comment:
This function is the `if (value[0].size() == 1)` branch and is otherwise unchanged.